packages/atomic/src/components/search-box-suggestions/atomic-search-box-recent-queries/atomic-search-box-recent-queries.tsx

```diff
@@ -0,0 +1,56 @@
+import {buildRecentQueriesList} from '@coveo/headless';
+import {Component, Element, Prop, State, h} from '@stencil/core';
+import {dispatchSearchBoxSuggestionsEvent} from '../suggestions-common';
+
+@Component({
+  tag: 'atomic-search-box-recent-queries',
+  shadow: true,
+})
+export class AtomicSearchBoxRecentQueries {
+  @Element() private host!: HTMLElement;
+
+  @State() public error!: Error;
+
+  @Prop() public maxWithQuery?: number;
+  @Prop() public maxWithoutQuery?: number;
+
+  componentWillLoad() {
+    try {
+      dispatchSearchBoxSuggestionsEvent(({engine, numberOfQueries}) => {
+        const recentQueriesList = buildRecentQueriesList(engine, {
+          // TODO: fetch initial state from cookies or local storage
+          initialState: {queries: ['hello', 'hola', 'bonjour', 'buongiorno']},
+          options: {maxLength: numberOfQueries},
+        });
+
+        return {
+          onInput: () => {},
```

Comment:
Should we show all recent queries in the suggestions or only those that include what we typed so far?

@johnnyCoveo

Reply:
It's in the mockups, when no query, we display them as is, when there is a query, we filter them out, you can refer to the part 2 for the implementation!